src/pages/workspaces/workspace/WorkspaceContainer.ts

```diff
@@ -124,6 +126,16 @@ export const WorkspaceContainer = (props: WorkspaceContainerProps) => {
   const isGoogleWorkspaceSyncing =
     workspaceLoaded && isGoogleWorkspace(workspace) && workspace?.workspaceInitialized === false;
 
+  // when the workspace refresh polling gets back an error for a workspace that is deleting
+  // redirect to list view
+  const handleWorkspaceError = (error: unknown) => {
+    if (error instanceof Response && error.status === 404) {
+      Nav.goToPath('workspaces');
+    }
+  };
+  // poll workspace state every 30 seconds
+  usePollingEffect(() => silentlyRefreshWorkspace(handleWorkspaceError), { ms: 30000, leading: false });
```

Comment:
@blakery I think we should change this so we are only polling on workspaces that are in the deleting state. I believe that when we update the workspace details, it also makes ajax calls to update runtimes and apps. And there is no reason for us to be doing this so aggressively for a workspace that is in any other state.

Reply:
+1, speculating this may be contributing to the spike in Leo traffic:
<img width="847" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/5368863/cac8e2ac-e642-43ea-a258-a5b8c86537db">